src/Mollie/WC/Gateway/AbstractSubscription.php

```diff
@@ -264,8 +273,20 @@ public function scheduled_subscription_payment( $renewal_total, WC_Order $renewa
 	        // Check that there is at least one valid mandate
 	        try {
 		        if ( $validMandate ) {
-			        Mollie_WC_Plugin::debug( $this->id . ': Valid mandate found for renewal order ' . $renewal_order_id );
 			        $payment = Mollie_WC_Plugin::getApiHelper()->getApiClient( $test_mode )->payments->create( $data );
+			        //update the valid mandate for this order
+                    if ((property_exists($payment, 'mandateId')
```

Comment:
I suggest to simply use `isset` here, because it does return `false` if the property exist but is `null`, while `property_exists` returns `true` and you need to check that is not null afterwards as in your current conditional.

so instead of:
`if ((property_exists($payment, 'mandateId')  && $payment->mandateId !== null)`

just: 
`isset($payment->mandateId)`